.github/workflows/muelu_clang_format.yml

```diff
@@ -0,0 +1,18 @@
+name: Check MueLu clang-format
+
+on: [pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: DoozyX/clang-format-lint-action@v0.16.2
+      with:
+        source: './packages/muelu/src'
+        exclude: './lib'
+        extensions: 'cpp,hpp'
+        clangFormatVersion: 8
```

Comment:
Why this version instead of anything more recent?

Reply:
Because Kokkos uses clang-format 8.0.0 (this workflow actually provides 8.0.1 when version 8 is specified)

Reply:
Does it really matter to have the same version as Kokkos? We should probably choose a version that is available on SEMS and from most modern distributions.

Reply:
I mean, we're both in Trilinos, so I'd hope that things are at least slightly similar/compatible.

The main problem is that clang-format is totally dependent on the version. It's horribly incompatible with itself. For example, if I run clang-format version 14.0.6 provided by SEMS, it disagrees with this PR resulting in
`523 files changed, 3841 insertions(+), 3836 deletions(-)`
Also, we can't guarantee that SEMS will support the same version of clang forever. Once the `sems-clang/14.0.6` module is gone, we're stuck in the same boat of running an unsupported version of `clang-format` anyway. What would we do in that case? Reformat all of MueLu again to agree with the supported module du jour?

Maybe I'm being a bit cynical, but while clang-format is great, different versions produce vastly different results, which is the main reason why I think we should point to some official LLVM release and not rely on a module, even though it may be slightly more difficult.

Reply:
If different versions produce different output we should just pick the most recent version in the hope that we can stick to it for the longest time possible and that there will be fewer and fewer changes for future versions.

Reply:
Okay. The latest version of LLVM that is released is 17.0.3, but the most recent version supported by SEMS TPLs is 14.0.6 (11.0.1 is also supported). However, the Github action only currently supports the following:
```
clang-format version 5.0.2 
clang-format version 6.0.1 
clang-format version 7.1.0 
clang-format version 8.0.1 
clang-format version 9.0.0
clang-format version 10.0.0
clang-format version 11.0.0
clang-format version 11.1.0
clang-format version 12.0.0
clang-format version 12.0.1
clang-format version 13.0.0
clang-format version 14.0.0
clang-format version 15.0.2
clang-format version 16.0.0
clang-format version 16.0.3
```
I could try to submit a PR to the workflow to add support for clang-format version 14.0.6 if there's a difference between 14.0.0 and 14.0.6 in MueLu.

Reply:
Let's table this and talk about it during the standup tomorrow.